zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceCancelProcessor.java

```diff
@@ -95,7 +94,8 @@ private boolean validateCommand(
         new AuthorizationRequest(
                 command,
                 AuthorizationResourceType.PROCESS_DEFINITION,
-                PermissionType.UPDATE_PROCESS_INSTANCE)
+                PermissionType.UPDATE_PROCESS_INSTANCE,
+                command.getValue().getTenantId())
```

Comment:
💭 In this way we might break user space, as the rejection changes. But IMO is more appropriate to return UNATHORIZED instead of NOT_FOUND

This is valid also for other processors

Reply:
Tenants are rather used for data isolation, so it should behave as if the objects of a tenant do not exist for users from other tenants

Reply:
Yeah, it's a good point 👍 We can argue to use FORBIDDEN instead of NOT_FOUND tho

Reply:
At the moment we have different behaviour, in the REST API we return 401 for unauthorized tenants, in the gRPC is 403. In the processors we reject the command with NOT_FOUND